packages/sanes-chrome-extension/src/logic/persona/persona.ts

```diff
@@ -0,0 +1,96 @@
+import { Amount } from '@iov/bcp';
+import { MultiChainSigner } from '@iov/core';
+
+import {
+  AccountManager,
+  AccountInfo,
+  createUserProfile,
+  AccountManagerChainConfig,
+} from '../user';
+import {
+  chainConnector,
+  getConfigurationFile,
+  codecTypeFromString,
+  algorithmForCodec,
+  pathBuilderForCodec,
+} from '../config';
+
+export class Persona {
+  /**
+   * Creates a new Persona instance.
+   *
+   * This function does everything that cannot be done in a constructor
+   * (because a constructor is synchonous): reading configs, connecting to the network,
+   * creating accounts.
+   */
+  public static async create(): Promise<Persona> {
```

Comment:
When I mentioned the Factory, the idea is having this constructor inside it. I like to have the factory because:
1. Is a standard way program against interfaces (no need to use new), and it is a pretty common way of solving this.
2. It is easier to put the logic of having multiple personas in the future.
3. You can put the logic of cleaning memory easily and solve one of the test problems at once.

```
class PersonaFactory {
  private static persona; // in the future personas: Map<id, Persona>

 public static create(): Persona { 
        .... 
       this.persona = persona;

      return this.persona;
  }

  public static get() {
     return this.persona
  }

  public static clear() { 
    this.persona = undefined;
  }

Reply:
Also, the fact of one entity know how to create itself seems to be odd, what is your reasoning behind this? Maybe I am missing something.

Reply:
> Also, the fact of one entity know how to create itself seems to be odd, what is your reasoning behind this? Maybe I am missing something.

The reason for putting the creator function into the `Persona` class is: it allows me to make the constructor private. This is important to avoid misuse of the constructor. The constructor has certain requirements for the parameters we put into it (in this case a shared profile instance) and by having exactly one place where this constructor is called, we can avoid misuse.

So `Persona.create()` is the way to create a single persona instance. Every additional funtionality, we can build around it in a separate class as soon as we need it (like the PersonaFactory idea).